xtable-core/src/main/java/org/apache/xtable/delta/DeltaSchemaExtractor.java

```diff
@@ -142,12 +144,24 @@ private DataType convertFieldType(InternalField field) {
     }
   }
 
+  private Metadata getMetaData(InternalType type) {
+    if (type == InternalType.UUID) {
+      return new MetadataBuilder().putString("xtableLogicalType", "uuid").build();
+    } else {
+      return Metadata.empty();
+    }
+  }
+
   public InternalSchema toInternalSchema(StructType structType) {
-    return toInternalSchema(structType, null, false, null);
+    return toInternalSchema(structType, null, false, null, null);
   }
 
   private InternalSchema toInternalSchema(
-      DataType dataType, String parentPath, boolean nullable, String comment) {
+      DataType dataType,
+      String parentPath,
+      boolean nullable,
+      String comment,
+      Metadata originalMetadata) {
```

Comment:
Add metadata as an input to better determine if a binary value is indeed a UUID in delta lake conversion